app/models/subscription.rb

```diff
@@ -51,11 +53,30 @@ def new_comment_by_someone_else?
   end
 
   def last_author
-    (discussion.comments.last || discussion).author
+    (last_comment || discussion).author
   end
 
-  def notify
-    NotificationsMailer.new_comment(id).deliver_later
+  def last_comment
+    discussion.comments.last
+  end
+
+  def notify_new_discussion
+    return unless new?
+    user.notifications.new_discussion(discussion)
```

Comment:
I'm not sure I understand that last comment

But anyway, I prefer to fix the LOD violation, actually. I'm just not sure how. Should I pass the user_id as an argument as well? (making them named arguments probably)